config/main.py

```diff
@@ -1357,12 +1357,67 @@ def apply_patch(ctx, patch_file_path, format, dry_run, ignore_non_yang_tables, i
 
+        results = {}
         config_format = ConfigFormat[format.upper()]
-        GenericUpdater().apply_patch(patch, config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+        if multi_asic.is_multi_asic():
+            # Initialize a dictionary to hold changes categorized by ASIC
+            changes_by_asic = {}
 
+            # Function to extract ASIC identifier from the change path
+            def extract_asic_id(path):
+                start = path.find("/") + 1
+                end = path.find("/", start)
+                return path[start:end], path[end:]  # Also return the modified path without ASIC ID
+
+            # Function to apply patch for a single ASIC.
+            def apply_patch_for_asic(asic_changes):
+                asic_id, changes = asic_changes
+
+                # Replace localhost to empty string which is db definition of Host
+                if asic_id.lower() == "localhost":
+                    asic_id = ""
+
+                try:
+                    # Call apply_patch with the ASIC-specific changes and predefined parameters
+                    GenericUpdater(namespace=asic_id).apply_patch(jsonpatch.JsonPatch(changes), config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+                    results[asic_id] = {"success": True, "message": "Success"}
+
+                    log.log_notice(f"'apply-patch' executed successfully for {asic_id} by {changes}")
+                except Exception as e:
+                    results[asic_id] = {"success": False, "message": str(e)}
+                    log.log_notice(f"'apply-patch' executed failed for {asic_id} by {changes} due to {str(e)}")
+
+            # Iterate over each change in the JSON Patch
+            for change in patch:
+                asic_id, modified_path = extract_asic_id(change["path"])
+
+                # Modify the 'path' in the change to remove the ASIC ID
+                change["path"] = modified_path
+
+                # Check if the ASIC ID is already in our dictionary, if not, initialize it
+                if asic_id not in changes_by_asic:
+                    changes_by_asic[asic_id] = []
+
+                # Add the modified change to the appropriate list based on ASIC ID
+                changes_by_asic[asic_id].append(change)
+
+            for asic_changes in changes_by_asic.items():
+                apply_patch_for_asic(asic_changes)
+
+            # Check if any ASIC updates failed
+            failures = [asic_id for asic_id, result in results.items() if not result['success']]
+
+            if failures:
+                failure_messages = '\n'.join([f"- {asic_id}: {results[asic_id]['message']}" for asic_id in failures])
+                raise Exception(f"Failed to apply patch on the following ASICs:\n{failure_messages}")
+
+        else:
+            GenericUpdater(multi_asic.DEFAULT_NAMESPACE).apply_patch(patch, config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
```

Comment:
>GenericUpdater [](http://example.com/codeflow?start=12&length=14)

Could you normalize the logic for both single-asic and multi-asic? single-asic is a special multi-asic with N=1.